lib/food_order.rb

```diff
@@ -0,0 +1,30 @@
+class FoodOrder
+  attr_accessor :basket, :takeaway_menu, :total_order
+  attr_reader :message
+
+  def initialize(takeaway_menu = Menu.new, message = TextClient.new)
+    @basket = []
+    @takeaway_menu = takeaway_menu
+    @message = message
+    @total_order = 0
+  end
+
+  def add_dish(dish)
+    selection = takeaway_menu.menu.select { |key, _value| key == dish }
+    basket << selection
+  end
```

Comment:
Here you are iterating through the menu instance variable of a different class. Whenever you reach into another class instance variable that makes your code more coupled. What if you wanted to change the way the menu is stored? You would have to change the FoodOrder class too. Instead, if you need something from a different class, it's good to create a method for it. Here you could have a method on `Menu` that returns a selection from a dish.
eg

in `FoodOrder`
```rb
def add_dish(dish)
    selection = takeaway_menu.get_selection(dish)
    basket << selection
end
```

in `Menu`
```rb
def get_selection(dish)
    @menu.select { |key, _value| key == dish }
end
```
This way, if you change the way `Menu` works later, as long as you keep the method `get_selection` working the same, then the FoodOrder should still work.
Decoupling like this will also make testing in isolation a lot easier as you just have to stub the return of the `get_selection` method instead of stubbing a full menu.